### DISCUSSION TITLE
On supporting an FRM "mixed" regime

### DISCUSSION THREAD

**Discussion by juanchossn** [nasa/HyperInSPACE#221] [Ideas]
It happens that some field OCR users have partial instrument-specific characterisation of their OCRs (i.e. characterised by only a reduced amount of effects). In that case, HyperCP still doesn't allow to correct for any effect, even those where instrument-specific characterisation exists (so it can only be operated in class based). This kind of hybrid cases may become more frequently as we ramp up to support new instruments into the HyperCP workflow. It may happen, e.g. for DALEC, that we may have, according to the characteristic, instrument-specific char. coeffs, class-based char. coeffs. or even nothing..

Having said this, there is interest to converge to a "unified/mixed/homogeneised" regime that will assess for each instrument and each characteristic whether there is instrument-sepcific info, clased-based info or nothing and will respectively correct for the effect + compute FRM unc, not correct + compute FRM uncertainty, or assess somehow the uncertainty (like the SIRREX approach for SeaBird).

This is one of the priorities for the coming FRM4SOC Opt. 2 extension.

**Comment by oceancolorcoder** [comment]
I agree we need to work toward a more flexible framework in which hybrid approaches like you describe can be accommodated so that partial characterizations can be applied with corrections where the specific instrument has been studied, with the fall-back of a class-based approach for other characterizations that were not performed. Implementation would be complex, as would describing the nature of the final product and product uncertainty depending on which characterizations were applied. Clearly, it would also need to be isolated in a new branch until complete or very mature to avoid disruption to dev and stable master releases. 

A difficult and related question is how to improve the L2 product uncertainties for TriOS with no absolute radiometric calibration (i.e., no class-based regime currently available). Keeping with your unified approach idea, this would exist somewhere on that spectrum (at the lowest, sub-Sirrex7 end).

**Comment by oceancolorcoder** [comment]
Development of a hybrid processing regime in which certain class-based characterizations are used as corrections rather than strictly for uncertainties is planned for the near term with the FRM4SOC initiative.
